worlds/checksmate/Options.py

```diff
@@ -0,0 +1,384 @@
+    """
+    display_name = "Fairy Chess Pawns"
+    option_vanilla = 0
+    option_mixed = 1
+    option_berolina = 2
+    default = 1
+
+
+class MinorPieceLimitByType(NamedRange):
+    """
+    How many of any given type of minor piece you might play with. If set to 1, you will never start with more than 1
+    Knight, nor 1 Bishop, but you may have both 1 Knight and 1 Bishop. If set to 0, this setting is disabled.
+    """
+    display_name = "Minor Piece Limit by Type"
+    range_start = 1
+    range_end = 9
+    default = 0
+    special_range_names = {
+        "disabled": 0,
+    }
+
+
+class MajorPieceLimitByType(NamedRange):
+    """
+    How many of any given type of major piece you might play with. If set to 1, you will never start with more than 1
+    Rook. If set to 0, this setting is disabled.
+    """
+    display_name = "Major Piece Limit by Type"
+    range_start = 1
+    range_end = 9
+    default = 0
+    special_range_names = {
+        "disabled": 0,
+    }
+
+
+class QueenPieceLimitByType(NamedRange):
+    """
+    How many of any given type of Queen-equivalent piece you might play with. If set to 1, you will never start with
+    more than 1 Queen. You may have both 1 Queen and 1 Amazon. If set to 0, this setting is disabled.
+    """
+    display_name = "Queen Piece Limit by Type"
+    range_start = 1
+    range_end = 5
+    default = 0
+    special_range_names = {
+        "disabled": 0,
+    }
+
+
+class PocketLimitByPocket(NamedRange):
+    """
+    How many Progressive Pocket items might be allocated to any given pocket. If this is set to 1, any given Pocket will
+    never hold anything more substantial than a Pawn. If this is set to 3, any given Pocket will never hold a Queen.
+
+    The default of 4 allows each of the 3 spaces to hold between 0-4 progressive items.
+    """
+    display_name = "Pocket Limit by Pocket"
+    range_start = 1
+    range_end = 4
+    default = 4
+    special_range_names = {
+        "disabled": 0,
+    }
+
+
+class QueenPieceLimit(NamedRange):
+    """
+    How many Queen-equivalent pieces you might play with. If set to 1, you will never have more than 1 piece upgraded to
+    a Queen. (This does nothing when greater than 'Queen Piece Limit by Type'.) You may still promote pawns during a
+    game. If set to 0, this setting is disabled.
+    """
+    display_name = "Queen Piece Limit"
+    range_start = 1
+    range_end = 7
+    default = 0
+    special_range_names = {
+        "disabled": 0,
+    }
+
+
+class LockedItems(ItemDict):
+    """
+    Guarantees that these progression and filler items will be unlockable.
+
+    Implementation note: Currently forces this many items into the item pool before distribution begins. This behaviour
+    is not guaranteed - a future version may simply validate the pool contains these items.
+    """
+    display_name = "Locked Items"
+
+
+class DeathLink(Toggle):
```

Comment:
was poking around in the apworld to debug something and came across it and wanted to comment while it was still fresh
you can import the generic deathlink option and just update the docstring like this (taken from Bomb Rush Cyberfunk: #2925 )
```
class BRCDeathLink(DeathLink):
    """When you die, everyone dies. The reverse is also true.
    This can be changed later in the options menu inside the Archipelago phone app."""
```

Reply:
Thanks, this is cool. However, I'm leaving my custom deathlink description, because "death" is not a prescribed action or state in Chess.